docs/src/adr/015adr-trace.md

```diff
@@ -230,11 +230,11 @@ as `CONSTANTS` in TLA+. For example, the field may look like:
   "params": [ "Missionaries", "Cannibals" ]
 ```
 
-The field `vars` is an array of names that must be set in every state.
+The field `vars` is an object, the keys of which are names that must be set in every state, and the values are string representations of their types (as defined in [ADR002]).
 For example, the field may look like:
 
 ```js
-  "vars": [ "bank_of_boat", "who_is_on_bank" ]
+  "vars": { "bank_of_boat": "Str", "who_is_on_bank": "Str -> Set(PERSON)" }
```

Comment:
Could "vars" carry more information in the future?
A more future-proof / backwards-compatible encoding would be

```suggestion
  "vars": [ { "name": "bank_of_boat", "type": "Str" },
            { "name": "who_is_on_bank", "type": "Str -> Set(PERSON)" } ]
```